v1/core_test.go

```diff
@@ -79,3 +82,37 @@ func TestHandlerWorksWithChiMux(t *testing.T) {
 	c.Assert(err, qt.IsNil)
 	c.Assert(len(out) > 0, qt.IsTrue)
 }
+
+// TestSwaggerJsonIgnoresAuthentication asserts that the `/swagger.json` endpoint
+// always respond with status code 200, without consulting with any authentication
+// middleware.
+func TestSwaggerJsonIgnoresAuthentication(t *testing.T) {
+	c := qt.New(t)
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	// A mock authenticator that always returns an authentication failure error,
+	// but it expects no calls. So, the test will fail if the mock struct method(s)
+	// get called.
+	authenticator := interfaces.NewMockAuthenticator(ctrl)
```

Comment:
question: where is this behavior described in the comment set?

Reply:
It's the `gomock` behaviour. When you create a new mock instance, you should then head out to set your expectations (e.g., by calling `authentiator.EXPECT(). ...`). If you don't set any expectation it means no calls are expected.

Reply:
ty